src/main/java/org/openrewrite/staticanalysis/RemoveUnusedLocalVariables.java

```diff
@@ -42,12 +44,38 @@ public class RemoveUnusedLocalVariables extends Recipe {
             example = "[unused, notUsed, IGNORE_ME]")
     String @Nullable [] ignoreVariablesNamed;
 
+    @Option(displayName = "Only remove variables of a given type",
+            description = "A fully qualified class name. Only unused local variables whose type matches this will be removed. " +
+                    "If empty or not set, all unused local variables are considered for removal.",
+            required = false,
+            example = "java.lang.String")
+    @Nullable
+    String withType;
+
     @Option(displayName = "Remove unused local variables with side effects in initializer",
             description = "Whether to remove unused local variables despite side effects in the initializer. Default false.",
             required = false)
     @Nullable
     Boolean withSideEffects;
 
+    @JsonCreator
+    public RemoveUnusedLocalVariables(
+            String @Nullable [] ignoreVariablesNamed,
+            @Nullable String withType,
+            @Nullable Boolean withSideEffects) {
+        this.ignoreVariablesNamed = ignoreVariablesNamed;
+        this.withType = withType;
+        this.withSideEffects = withSideEffects;
+    }
+
+    @InlineMe(replacement = "new RemoveUnusedLocalVariables(ignoreVariablesNamed, null, withSideEffects)")
```

Comment:
The contract is that the value corresponds exactly to the body, so `this(...)` rather than `new RemoveUnusedLocalVariables(...)`.